### PR TITLE
Zhihao: add view.broadcast feature

### PR DESCRIPTION
**Pull request by chenzhihao**
Add broadcast feature to broadcast events from Parent to Sub view which can bubble down.

### CODE REVIEW

File: b2.js [hunk 1]
```diff
@@ -153,6 +172,20 @@
...
       });
 
+      component.listenTo(this,'__broadcast__', function () {
+        var args = [].slice.apply(arguments);
+        var eventName = args.shift();
+        if (component.broadcastEvents.hasOwnProperty(eventName)) {
+          var funcName = component.broadcastEvents[eventName];
+          var func = component[funcName];
+          if (_.isFunction(func)) {
+            func.apply(component,args);
```
Comment (blacktail) [inline b2.js]: please add space after the comma

File: b2.js [hunk 2]
```diff
@@ -153,6 +172,20 @@
...
+          } else {
+            component.broadcast.apply(component,arguments);
```
Comment (blacktail) [inline b2.js]: please add space after the comma

File: b2.js [hunk 3]
```diff
@@ -82,6 +82,25 @@
...
+    },
+
+    // broadcast events to subviews
+    // the format is
+    // view.broadcast(eventName, arg1, arg2...);
+    //
+    broadcast: function (){
+      var args = [].slice.apply(arguments);
+      args.unshift('__broadcast__');
+      this.trigger.apply(this, args);
```
Comment (blacktail) [inline b2.js]: no need using additional unshift, you can just write list this:

```
this.trigger.apply(this, [].slice.call(arguments, 1));
```

**Comment by chenzhihao** [comment]
code is format now.

File: b2.js [hunk 4]
```diff
@@ -153,6 +172,20 @@
...
+          var funcName = component.broadcastEvents[eventName];
+          var func = component[funcName];
+          if (_.isFunction(func)) {
+            func.apply(component,args);
+          } else {
+            component.broadcast.apply(component,arguments);
+          }
+        }
+      });
+
```
Comment (Cohlint) [inline b2.js]: good job

**Pull request merged**